src/rpcblockchain.cpp

```diff
@@ -1593,25 +1394,16 @@ UniValue currentcontractaverage(const UniValue& params, bool fHelp)
 
     UniValue res(UniValue::VOBJ);
 
-    std::string contract = NN::GetInstance()->GetNeuralContract();
-    double out_beacon_count = 0;
-    double out_participant_count = 0;
-    double out_avg = 0;
-    double avg = GetSuperblockAvgMag(contract, out_beacon_count, out_participant_count, out_avg, false, nBestHeight);
-    bool bValid = VerifySuperblock(contract, pindexBest);
-    //Show current contract neural hash
-    std::string sNeuralHash = NN::GetInstance()->GetNeuralHash();
-    std::string neural_hash = GetQuorumHash(contract);
-
-    res.pushKV("Contract", contract);
-    res.pushKV("avg", avg);
-    res.pushKV("beacon_count", out_beacon_count);
-    res.pushKV("avg_mag", out_avg);
-    res.pushKV("beacon_participant_count", out_participant_count);
-    res.pushKV("superblock_valid", bValid);
-    res.pushKV(".NET Neural Hash", sNeuralHash.c_str());
-    res.pushKV("Length", (int)contract.length());
-    res.pushKV("Wallet Neural Hash", neural_hash);
+    const NN::Superblock superblock = NN::GetInstance()->GetSuperblockContract();
+
+    res.pushKV("Contract", SuperblockToJson(superblock));
+    res.pushKV("beacon_count", superblock.m_cpids.TotalCount());
```

Comment:
This fails on 32-bit because superblock.m_cpids.TotalCount() is size_t, but univalue doesn't understand 32-bit integers. Only int and int64_t. You will have to cast them. That is why the 32-bit Travis builds are failing.

Reply:
That is also why I try so hard to keep the multi-arch builds in Travis in working order... so we can detect stuff like this.

Reply:
You can cherry-pick https://github.com/jamescowens/Gridcoin-Research/commit/8f9de6c6b364f1afa95cc028d1ca4a01e7b0f539 from my tallyoverhaul branch for the fix.

Reply:
Thanks for taking the time to look into that and rewrite the code. I think I already squashed fixes for each of these back into a previous commit that's waiting on some other tweaks before I force-push. Have not had time to come back and finish that because of a special family event 🙂